docs/reference/generated/scroll-area-root.json

```diff
@@ -11,7 +11,26 @@
       "description": "Allows you to replace the component’s HTML element\nwith a different tag, or compose it with another component.\n\nAccepts a `ReactElement` or a function that returns the element to render."
     }
   },
-  "dataAttributes": {},
+  "dataAttributes": {
+    "data-has-overflow-x": {
+      "description": "Present when the scroll area has horizontal overflow."
+    },
+    "data-has-overflow-y": {
+      "description": "Present when the scroll area has vertical overflow."
+    },
+    "data-overflow-x-end": {
+      "description": "Present when there is overflow on the inline end side for the horizontal axis."
+    },
+    "data-overflow-x-start": {
+      "description": "Present when there is overflow on the inline start side for the horizontal axis."
+    },
+    "data-overflow-y-end": {
+      "description": "Present when there is overflow on the block end (bottom) side."
+    },
+    "data-overflow-y-start": {
+      "description": "Present when there is overflow on the block start (top) side."
+    }
+  },
```

Comment:
Present when `.scrollTop/Left > 0`

I'm curious if there should also be CSS variables for the scroll amount if you want to transition the strength of e.g. an overflow gradient by the amount. Maybe CSS scroll-driven animations already solve this, though